test/reductions/reduceInIfExpr.chpl

```diff
@@ -0,0 +1,2 @@
+var A: [1..10] int = 42;
+writeln(if (+ reduce [a in A] a) == 10 * 42 then 'Pass' else 'Fail');
```

Comment:
I'd also be inclined to check in the other test Greg showed in #15171 - its behavior hopefully shouldn't have changed and I don't think there's harm having more tests to ensure things don't backslide